File: src/main/java/com/david/knowledgegraph/Node.java
package com.david.knowledgegraph;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;


public class Node
{
  private int IDLENGTH = 6;
  public ArrayList<String> children;
  public ArrayList<String> parents;
  public String Id;
  public String text = "";
  public int level;

  private void setId() {
    String h = Integer.toString(this.text.hashCode());
    String Id;

    if (this.IDLENGTH > h.length()) {
      this.Id = h;
    } else {
      this.Id = h.substring(0, this.IDLENGTH);
    }
  }

  public Node(String text) {
    String textLines[] = text.split(" ");
    List<String> splitList = Arrays.asList(textLines);
    List<String> subList= splitList.subList(1, textLines.length);
    for (String s : subList) {
      this.text += s.toString();
      this.text += " ";
    }
    String levelLength = textLines[0];
    this.level = levelLength.length();
    setId();

    this.children = new ArrayList<String>();
    this.parents = new ArrayList<String>();
  }

  public void addParent(String parentId) {
    this.parents.add(parentId);
  }

  public void addChild(String childId) {
    this.children.add(childId);
  }

  @Override
  public String toString() {
    return "Node:" + this.Id;
  }

}

File: src/main/scala/knowledgegraph/Node.scala
package knowledgegraph

import scala.collection.mutable.ArrayBuffer


class Node(_text: String) {
  val children = ArrayBuffer[String]()
  val parents = ArrayBuffer[String]()
  val IDLENGTH: Int = 6

  var Id: String = ""
  var level: Int = 0
  var text: String = ""

  processText(_text)

  private def processText(text: String): Unit = {
    val textLines: Array[String] = text.split(" ")
    this.text = textLines.slice(1, textLines.length).mkString(" ")
    this.level = textLines(0).length
    setId()
  }

  private def setId() =
  {
    val h: String = this.text.hashCode().toString

    if (this.IDLENGTH > h.length) {
      this.Id = h
    } else {
      this.Id = h.substring(0, this.IDLENGTH)
    }
  }

  def addParent(parentId: String): Unit =
  {
    this.parents += parentId
  }

  def addChild(childId: String): Unit = {
    this.children += childId
  }

  override def toString = {
    "Node:" + this.Id
  }

}
